vault/expiration.go

```diff
@@ -210,6 +219,13 @@ func (r *revocationJob) OnFailure(err error) {
 
 	if pending.revokesAttempted >= maxRevokeAttempts {
 		r.m.logger.Trace("lease has consumed all retry attempts", "lease_id", r.leaseID)
+		le, loadErr := r.m.loadEntry(r.nsCtx, r.leaseID)
+		if loadErr != nil {
+			r.m.logger.Warn("failed to mark lease as zombie - failed to load", "lease_id", r.leaseID, "err", loadErr)
+			return
+		}
+
+		r.m.markLeaseAsZombie(r.nsCtx, le, errors.New("lease has consumed all retry attempts"))
```

Comment:
If we're just going to be using the same error message for each lease, it will be cheaper to just set a boolean instead. Do you think we should switch?

Reply:
hmm, what would this look like? I'm all for cheaper - even if that means just shortening the string in the error. good suggestion!

Reply:
https://hashicorp.atlassian.net/browse/VAULT-2303